docs/reference/mapping/types/runtime.asciidoc

```diff
@@ -0,0 +1,94 @@
+[[runtime]]
+=== Runtime
+
+////
+[source,console]
+----
+PUT /my-index
+----
+// TESTSETUP
+////
+
+Typically, you must index fields to {es} before they can be retrieved,
+aggregated, or searched. With runtime fields, you can explicitly define a field
+in the mapping and access it at runtime without indexing your data. This
+flexibility allows you to more quickly ingest raw data into the Elastic Stack
+and immediately access it. By dynamically evaluating runtime fields at query
+time, you can choose which fields to index and optimize disk space.
+
+include::{es-ref-dir}/mapping/types.asciidoc[tag=runtime-fields-description]
+
+// Each runtime field is of the `runtime` data type, and has its own
+// <<mapping-types,field type>>, such as `boolean`, `long`, or `keyword`. The
+// field type identifies the data type In
+// the following example, the data type is `runtime` and the runtime field type is
+// `keyword`.
+
+==== Specifying runtime fields
+Runtime fields do not have access to the
+<<mapping-source-field,`_source` field>>. You specify runtime fields in the
+mapping by <<modules-scripting-using,defining a script>>, which always has
+access to the `source` field. At search time, the script runs and generates
+values for each scripted field.
+
+The script in the following request extracts the day of the week from the
+`timestamp` field, which is defined as a `date` data type.
+
+[source,console]
+----
+PUT /my-index/_mappings
+{
+  "properties" : {
+    "day_of_week" : {
+      "type" : "runtime", <1>
+      "runtime_type" : "keyword", <2>
+      "script" : {
+        "source" : "emit(doc['timestamp'].value.dayOfWeekEnum.getDisplayName(TextStyle.FULL, Locale.ROOT))"
+      }
+    }
+  }
+}
+----
+
+<1> Runtime fields are of the `runtime` data type.
+<2> Each runtime has its own field type, defined by `runtime_type`.
+
+[[runtime-params]]
+==== Parameters for `runtime` fields
+Runtime fields accept the following parameters:
+
+`type`::
+The type of runtime computation to perform at query time. Currently, runtime
+fields only support the `runtime` data type.
+
+`runtime_type`::
+The <<mapping-types,field type>> for each scripted field. {es}
+supports `boolean`, `date`, `double`, `ip`, `keyword`, and `long`.
+
+==== Retrieving runtime fields
+Because runtime fields are not part of the `_source` field, they are not
+returned in search hits by default. You can request runtime fields by using the
+<<search-fields,`fields`>> parameter, or by using the
+<<docvalue-fields,`docvalue_fields`>> parameter to return `doc_values` for one
+or more fields in the search response.
+
+Use the <<search-field-caps,field capabilities API>> to return runtime fields
+like any other field. For example, a runtime field with a `runtime_type` of
+`keyword` returns as any other field that belongs to the `keyword` family.
```

Comment:
I'm not sure I'd add this paragraph to the docs at all, or at least, not with "retrieving". It is an important bit of information if you are building an application that consumes Elasticsearch and is super general. I think that is, mostly Kibana. My gut feeling is that most applications that consume Elasticsearch manage the mapping themselves and don't typically use the field capabilities API at all. If that is true, then I think this bit of information belongs as a note on the field caps api page.

Reply:
I'm moving this paragraph to the field caps API page, with a link back to the runtime fields page.